pkg/eval/compile_effect.go

```diff
@@ -142,12 +138,7 @@ func (op *pipelineOp) exec(fm *Frame) Exception {
 				msg += ", errors = " + err.Error()
 			}
 			if fm.Evaler.getNotifyBgJobSuccess() || err != nil {
-				editor := fm.Evaler.Editor()
-				if editor != nil {
-					editor.Notify("%s", msg)
-				} else {
-					fm.ErrorFile().WriteString(msg + "\n")
-				}
+				fm.ErrorFile().WriteString(msg + "\n")
```

Comment:
You were probably trying to clean up the evaler so that it no longer needs to contain a reference to the editor; cleanups should not be done at the expense of real features. Removing the notification logic is a regression.

Reply:
There is currently no way for `fm.Evaler.Editor()` to return a non-nil value. Both in actual use and unit tests. The presence of this unused interface method greatly complicated the implementation of this change for zero benefit since the added functions were simply stubs that did nothing.

Reply:
Also, it's not at all obvious that we should, or event want to, automatically redirect/capture the output of background jobs. The POSIX standard handles this via tty job control; e.g., `SIGTTIN` and `SIGTTOU` sent to a background job that is not in control of the tty. There is discussion in issue #410 about whether Elvish should fully implement POSIX job control. Regardless, I think there is no good argument for retaining the existing, unused, notification mechanism since doing so increases the complexity of this change for zero benefit today, and probably tomorrow. This background job notification mechanism can always be reintroduced if, and when, there is consensus it is needed and how it should work.

Reply:
Huh, OK, I didn't realise this functionality has been broken. While I'd prefer you keep it since I do plan to fix the feature, that does mean removing it is harmless now.